coala_quickstart/generation/SettingsFilling.py

```diff
@@ -165,3 +168,93 @@ def resolve_anomaly(setting_name,
                                                    join_names(values)),
                   REPORT_ANOMALY_COLOR))
     return input()
+
+
+def require_setting(setting_name, setting_info, section):
+    """
+    This method is responsible for prompting a user about a missing setting and
+    taking its value as input from the user.
+
+    :param setting_name: Name of the setting missing
+    :param setting_info: A dictionary of the form
+
+    ::
```

Comment:
huh, I didn't know we did codeblocks this way - there's a bunch of docs that's strewn all over the code base without this

oh well, now's a good time to start as any